src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzerTests.cs

```diff
@@ -54,6 +54,10 @@ public static IEnumerable<object[]> Create_DifferentTfms()
             yield return new object[] { "build_property.TargetFramework = net5.0-ios14.0", true };
             yield return new object[] { "build_property.TargetFramework = Net99", true };
             yield return new object[] { "build_property.TargetFramework = netcoreapp5", false };
+            yield return new object[] { "build_property.TargetFramework = net10.0", true };
+            yield return new object[] { "build_property.TargetFramework = net10", true };
+            yield return new object[] { "build_property.TargetFramework = nonesense\nbuild_property.TargetFrameworkIdentifier=.NETCoreApp\nbuild_property.TargetFrameworkVersion=v11", true };
+            yield return new object[] { "build_property.TargetFramework = nonesense\nbuild_property.TargetFrameworkIdentifier=.NETCoreApp\nbuild_property.TargetFrameworkVersion=v11.0", true };
```

Comment:
Those extra props should in theory be set by `Microsoft.CodeAnalysis.NetAnalyzers.targets` which gets generated from `MSBuildPropertyOptionNames`. I've validated it did but not sure if there is easy way to force tests to also do it. Possibly you can invoke msbuild target in the tests. Would be nice to have something closer to E2E in the tests

Reply:
Tanner is testing whether custom TFM aliases (i.e. `nonsense`) work correctly. Custom aliases need the `TargetFrameworkIdentifier` and the `TargetFrameworkVersion` explicitly set as the SDK can't infer those from the alias.

These two test cases are good as they are and valuable to have.

Reply:
The test fix to get CI passing is going to require all of these setting `TargetFrameworkIdentifier` and `TargetFrameworkVersion`. It is "best" to also set and validate `TargetFramework` as would be done from a proper build to help catch any other logic that is trying to use it rather than the more proper metadata that exists.